engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/SourceMetricsFunction.scala

```diff
@@ -22,3 +23,20 @@ private[registrar] class SourceMetricsFunction[T](sourceId: String) extends Proc
   }
 
 }
+
+private[registrar] class TestSourceMetricsFunction[T](sourceId: String) extends ProcessFunction[T, T] {
```

Comment:
DRY. Please try to pass `EngineRuntimeContext` here. We did it in other places - take a look at things like `convertToEngineRuntimeContext: RuntimeContext => EngineRuntimeContext`. In case if it will be too hard to achieve, you can always extract this metric provider creation logic to one function.